src/mgr/MgrStandby.cc

```diff
@@ -317,29 +317,8 @@ void MgrStandby::respawn()
   }
   new_argv[orig_argc] = NULL;
 
-  /* Determine the path to our executable, test if Linux /proc/self/exe exists.
-   * This allows us to exec the same executable even if it has since been
-   * unlinked.
-   */
-  char exe_path[PATH_MAX] = "";
-  if (readlink(PROCPREFIX "/proc/self/exe", exe_path, PATH_MAX-1) == -1) {
-    /* Print CWD for the user's interest */
-    char buf[PATH_MAX];
-    char *cwd = getcwd(buf, sizeof(buf));
-    ceph_assert(cwd);
-    dout(1) << " cwd " << cwd << dendl;
-
-    /* Fall back to a best-effort: just running in our CWD */
-    strncpy(exe_path, orig_argv[0], PATH_MAX-1);
-  } else {
-    dout(1) << "respawning with exe " << exe_path << dendl;
-    strcpy(exe_path, PROCPREFIX "/proc/self/exe");
-  }
-
-  dout(1) << " exe_path " << exe_path << dendl;
-
   unblock_all_signals(NULL);
-  execv(exe_path, new_argv);
```

Comment:
can we use `stat()` with the `orig_argv[0]` if the file exists, we can happily use the `orig_argv[0]`, otherwise,  fallback to `readlink(PROCPREFIX "/proc/self/exe", exe_path, PATH_MAX-1)`, and then `strcpy(exe_path, PROCPREFIX "/proc/self/exe")` , again use `orig_argv[0]` as a fallback in case `readlink()` fails.